flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/reader/InfluxDBSplitReader.java

```diff
@@ -119,35 +135,60 @@ public void handle(final HttpExchange t) throws IOException {
             final BufferedReader in =
                     new BufferedReader(
                             new InputStreamReader(t.getRequestBody(), StandardCharsets.UTF_8));
-            String line;
+
             try {
+                String line;
+                final List<DataPoint> points = new ArrayList<>();
+                int n = 0;
                 while ((line = in.readLine()) != null) {
-                    final DataPoint dataPoint = LineProtocolParser.parse(line);
-                    if (dataPoint != null) {
-                        InfluxDBSplitReader.this.ingestionQueue.put(dataPoint);
+                    final DataPoint dataPoint =
+                            InfluxDBSplitReader.this.parser.parseToDataPoint(line);
+                    points.add(dataPoint);
+                    n++;
+                    if (n > MAXIMUM_LINES_PER_REQUEST) {
+                        throw new RequestTooLargeException(
+                                "Payload too large. Maximum number of lines per request is "
+                                        + MAXIMUM_LINES_PER_REQUEST
+                                        + ".");
                     }
                 }
-            } catch (final InterruptedException e) {
-                // TODO: check what to do with failing put to queue
+
+                InfluxDBSplitReader.this.ingestionQueue.put(points); // TODO use offer with timeout
+
+                t.sendResponseHeaders(204, -1);
+            } catch (final ParseException e) {
+                final byte[] response = e.getMessage().getBytes();
+                // 400 Bad Request
+                t.sendResponseHeaders(400, response.length);
+                OutputStream os = t.getResponseBody();
+                os.write(response);
+                os.close();
+            } catch (final RequestTooLargeException e) {
+                final byte[] response = e.getMessage().getBytes();
+                // 413 Payload Too Large
+                t.sendResponseHeaders(413, response.length);
+                OutputStream os = t.getResponseBody();
+                os.write(response);
+                os.close();
+            } catch (
+                    final InterruptedException
+                            e) { // TODO InterruptedException may not be the right type
+                final byte[] response = "Server overloaded".getBytes();
+                // 429 Too Many Requests
+                t.sendResponseHeaders(429, response.length);
+                OutputStream os = t.getResponseBody();
+                os.write(response);
+                os.close();
+
+                // TODO rethrow/forward exception so that Flink knows the ingestion queue was full
```

Comment:
Since Java 7 you can use this:
```Java
catch(IOException | SQLException ex){
     logger.error(ex);
     throw new MyException(ex.getMessage());
}
```
So catching multiple exceptions at once. Besides the logic inside each catch is almost the same. Consider putting them in a method.

Reply:
Not applicable here because of different responses. Though, we can implement a private response function as code is duplicated.